### PR TITLE
session: make plan replayer capture support prepared stmt

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/40161

Problem Summary:

### What is changed and how it works?

make plan replayer capture support prepared stmt

### Check List

Tests 

- [x] Unit test

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: session/session.go [hunk 1]
```diff
@@ -2372,6 +2373,17 @@ func runStmt(ctx context.Context, se *session, s sqlexec.Statement) (rs sqlexec.
...
+		stmtNode := s.GetStmtNode()
+		startTS := se.GetSessionVars().StmtCtx.StmtSnapshotTS
+		if se.GetSessionVars().EnablePlanReplayedContinuesCapture {
+			if checkPlanReplayerContinuesCaptureValidStmt(stmtNode) {
+				checkPlanReplayerContinuesCapture(se, stmtNode, startTS)
+			}
+		} else {
+			checkPlanReplayerCaptureTask(se, stmtNode, startTS)
+		}
+	}
```
Comment (qw4990) [inline session/session.go:2386]: Can we add this logic into `ExecStmt.FinishExecuteStmt`?

Comment (Yisaer) [reply session/session.go]: updated.

**Review by qw4990 — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**